src/util/benchmark.cc

```diff
@@ -0,0 +1,198 @@
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS    //
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,          //
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF    //
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.               //
+ */
+
+
+#include "util/benchmark.h"
+
+#include "core/util/exception.h"
+#include "core/util/log.h"
+
+
+namespace bpo = boost::program_options;
+void Benchmark::do_DSA()
+{
+  LOG(info) << "--------DSA---------";
+  do_benchmark<kovri::core::DSAVerifier, kovri::core::DSASigner>(
+      Benchmark::BENCHMARK_COUNT,
+      kovri::core::DSA_PUBLIC_KEY_LENGTH,
+      kovri::core::DSA_PRIVATE_KEY_LENGTH,
+      kovri::core::DSA_SIGNATURE_LENGTH,
+      kovri::core::CreateDSARandomKeys);
+}
+void Benchmark::do_ECDSAP256()
+{
+  LOG(info) << "-----ECDSAP256------";
+  do_benchmark<kovri::core::ECDSAP256Verifier, kovri::core::ECDSAP256Signer>(
+      Benchmark::BENCHMARK_COUNT,
+      kovri::core::ECDSAP256_KEY_LENGTH,
+      kovri::core::ECDSAP256_KEY_LENGTH / 2,
+      kovri::core::ECDSAP256_KEY_LENGTH,
+      kovri::core::CreateECDSAP256RandomKeys);
+}
+void Benchmark::do_ECDSAP384()
+{
+  LOG(info) << "-----ECDSAP384------";
+  do_benchmark<kovri::core::ECDSAP384Verifier, kovri::core::ECDSAP384Signer>(
+      Benchmark::BENCHMARK_COUNT,
+      kovri::core::ECDSAP384_KEY_LENGTH,
+      kovri::core::ECDSAP384_KEY_LENGTH / 2,
+      kovri::core::ECDSAP384_KEY_LENGTH,
+      kovri::core::CreateECDSAP384RandomKeys);
+}
+void Benchmark::do_ECDSAP521()
+{
+  LOG(info) << "-----ECDSAP521------";
+  do_benchmark<kovri::core::ECDSAP521Verifier, kovri::core::ECDSAP521Signer>(
+      Benchmark::BENCHMARK_COUNT,
+      kovri::core::ECDSAP521_KEY_LENGTH,
+      kovri::core::ECDSAP521_KEY_LENGTH / 2,
+      kovri::core::ECDSAP521_KEY_LENGTH,
+      kovri::core::CreateECDSAP521RandomKeys);
+}
+void Benchmark::do_EDDSS25519()
+{
+  LOG(info) << "-----EDDSA25519-----";
+  do_benchmark<kovri::core::EDDSA25519Verifier, kovri::core::EDDSA25519Signer>(
+      Benchmark::BENCHMARK_COUNT,
+      kovri::core::EDDSA25519_PUBLIC_KEY_LENGTH,
+      kovri::core::EDDSA25519_PRIVATE_KEY_LENGTH,
+      kovri::core::EDDSA25519_SIGNATURE_LENGTH,
+      kovri::core::CreateEDDSARandomKeys);
+}
+Benchmark::Benchmark() : m_Desc("Options")
+{
+  m_Desc.add_options()("help,h", "produce this help message")
+     ("DSA,d", bpo::bool_switch()->default_value(false), "DSA")
+     ("ECDSAP256,2", bpo::bool_switch()->default_value(false), "ECDSAP256 test")
+     ("ECDSAP384,3", bpo::bool_switch()->default_value(false), "ECDSAP384 test")
+     ("ECDSAP521,5", bpo::bool_switch()->default_value(false), "ECDSAP521 test")
+     ("EDDSA25519,e", bpo::bool_switch()->default_value(false),
+      "EDDSA25519 test")
+     ("all,a", bpo::bool_switch()->default_value(false), "all tests");
```

Comment:
This is all nice but, because of https://github.com/monero-project/kovri/issues/498, this was all just extra work for the task at hand.